transit_odp/site_admin/exports.py

```diff
@@ -118,6 +118,24 @@ def get_ppc_weekly_overall(organisation_id: int) -> float:
 ppc_overall_score = make_ppc_weekly_overall()
 
 
+def make_avl_compliance_status():
+    avl_compliance_status = {}
+
+    def get_avl_compliance_status(organisation_id: int):
+        if organisation_id not in avl_compliance_status.keys():
+            avl_compliance_status[organisation_id] = (
+                AVLDataset.objects.get_active()
+                .add_avl_compliance_status_cached()
+                .filter(organisation_id=organisation_id)
+            )
+        return avl_compliance_status[organisation_id]
+
+    return get_avl_compliance_status
+
+
+avl_compliance_status_cached_caller = make_avl_compliance_status()
```

Comment:
Yes It's a closure, but a Python closure, if you look at row 98 there is another closure.
Here I use the same logic. What I'd avoid here is hammering the db.

Reply:
It's similar to a decorator

Reply:
Yeah sure not a problem  to initialize in the constructor of the `DatasetPublishingCSV`, 
but since this is a bug I'd prefer leave the code already been approved and in `dev` I mean 
I think it's better not touch the code from row `98`, unless you don't want create a new bug story.